test/unit/math/laplace/wolfe_line_search_test.cpp

```diff
@@ -0,0 +1,830 @@
+  Eigen::VectorXd newton = Q.ldlt().solve(b);
+  Eigen::VectorXd curr = prev + newton;
+
+  WolfeInfo info = harness.make_info(obj, prev, curr);
+  info.curr_.alpha() = 0.5;
+  WolfeInfo before = info;
+
+  auto status = harness.run(info, obj);
+  EXPECT_EQ(status.stop_, WolfeReturn::Wolfe)
+      << "Expected Wolfe but wolfe returned "
+      << stan::math::internal::wolfe_status_str(status);
+  EXPECT_EQ(status.num_backtracks_, 0);
+
+  auto [p, dir0] = initial_direction(obj, before);
+  double phi0 = obj(before.prev_.a(), before.prev_.theta());
+  double alpha = info.curr_.alpha();
+  double phi_alpha = obj(info.curr_.a(), info.curr_.theta());
+  EXPECT_GE(phi_alpha, phi0 + harness.opt.c1 * alpha * dir0 - 1e-12);
+}
+
+// Checks that the search zooms to satisfy the curvature condition.
+TEST(WolfeLineSearch, RequiresZoomForCurvature) {
+  const int n = 2;
+  LineSearchHarness harness(n);
+  Eigen::MatrixXd Q = make_spd_from_diag({0.4, 4.0});
+  Eigen::VectorXd b(n);
+  b << 0.6, 1.5;
+  QuadraticObjective obj(Q, b);
+
+  Eigen::VectorXd prev = Eigen::VectorXd::Zero(n);
+  Eigen::VectorXd newton = Q.ldlt().solve(b);
+  Eigen::VectorXd curr = prev + newton;
+
+  WolfeInfo info = harness.make_info(obj, prev, curr);
+  info.curr_.alpha() = 1.0;  // initial doubling overshoots the maximiser
+  WolfeInfo before = info;
+
+  auto status = harness.run(info, obj);
+  EXPECT_EQ(status.stop_, WolfeReturn::Wolfe)
+      << "Expected Wolfe but wolfe returned "
+      << stan::math::internal::wolfe_status_str(status);
+  EXPECT_GE(status.num_backtracks_, 0);
+
+  auto [p, dir0] = initial_direction(obj, before);
+  double dir_alpha = directional_derivative(obj, info.curr_.a(), p);
+  EXPECT_LE(std::abs(dir_alpha), harness.opt.c2 * std::abs(dir0) + 1e-12);
+}
+
+// Checks that exact Armijo equality is accepted.
+TEST(WolfeLineSearch, ArmijoEqualityAccepted) {
+  LineSearchHarness harness(1);
+  harness.opt.c1 = 0.6;
+  Eigen::MatrixXd Q = make_spd_from_diag({1.6});
+  Eigen::VectorXd b(1);
+  b << 1.0;
+  QuadraticObjective obj(Q, b);
+
+  Eigen::VectorXd prev = Eigen::VectorXd::Zero(1);
+  Eigen::VectorXd curr = Eigen::VectorXd::Ones(1);
+  WolfeInfo info = harness.make_info(obj, prev, curr);
+  info.curr_.alpha() = 0.25;  // trial alpha -> 0.5
+  WolfeInfo before = info;
+
+  auto status = harness.run(info, obj);
+  EXPECT_EQ(status.stop_, WolfeReturn::Wolfe)
+      << "Expected Wolfe but wolfe returned "
+      << stan::math::internal::wolfe_status_str(status);
+
+  auto [p, dir0] = initial_direction(obj, before);
+  double alpha = info.curr_.alpha();
+  EXPECT_NEAR(alpha, 0.5, 1e-12);
+  double phi0 = obj(before.prev_.a(), before.prev_.theta());
+  double phi_alpha = obj(info.curr_.a(), info.curr_.theta());
+  double armijo_gap = phi_alpha - (phi0 + harness.opt.c1 * alpha * dir0);
+  EXPECT_NEAR(armijo_gap, 0.0, 1e-12);
+}
+
+// Tests stay focused on the strong-Wolfe variant because the API does not yet
+// expose a weak-Wolfe configuration.
```

Comment:
Ok... I lost track of the difference between the strong-Wolfe and weak-Wolfe configuration. Can you give me a reminder.